### PR TITLE
[4] add missing and fix wrong includes in com_finder, and remove unset on undefined var

### PR DESCRIPTION
**Pull request by PhilETaylor**
Code review

Add missing includes for docblock type hints

Correct incorrect type hint and include that too (FilterTable)

Remote `$tokens` from the `unset` as `$tokens` is not even set in this method and undefined at this time and place.

### CODE REVIEW

File: administrator/components/com_finder/src/Indexer/Indexer.php [hunk 1]
```diff
@@ -18,7 +18,9 @@
 use Joomla\CMS\Plugin\PluginHelper;
 use Joomla\CMS\Profiler\Profiler;
 use Joomla\Database\ParameterType;
+use Joomla\Database\QueryInterface;
 use Joomla\String\StringHelper;
+use Exception;
```
Comment (Quy) [inline administrator/components/com_finder/src/Indexer/Indexer.php]: Please move to line 14.

Comment (PhilETaylor) [reply administrator/components/com_finder/src/Indexer/Indexer.php]: haha that's where phpStorm put it - pushing now

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on ec614a95eaa53d6d159f0261c1158d07b563f433<br /><br />Tested successfully in Beta7-dev of 16 January.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32058">issues.joomla.org/tracker/joomla-cms/32058</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on ec614a95eaa53d6d159f0261c1158d07b563f433<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32058">issues.joomla.org/tracker/joomla-cms/32058</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**